core-blocks/paragraph/index.js

```diff
@@ -236,14 +236,12 @@ class ParagraphBlock extends Component {
 								if ( after ) {
 									blocks.push( createBlock( name, { content: after } ) );
 								}
+								const replaceArrayStart = before ?
+									[ createBlock( name, { ...attributes, content: before } ) ] :
+									[];
```

Comment:
There's a subtle difference here with the previous behavior. We're creating a new block with a new UID which resets all other properties of the blocks and cause a remount instead of a rerender. Maybe it's fine, just noting.

Reply:
That's true we change the uid and we have a remount but I think doing that is ok.

Reply:
Could we reuse the block that is being replaced in the reducer if the name is the same?

Reply:
We had already an onReplace in the paragraph with logic I just added one line there that reuses to UID, I think it simplifies the things and removes the need for any TinyMCE hacks.